### ISSUE TITLE
ValueNode.toValueNode() is not converting to JsonNode

### ISSUE THREAD

**Issue by kiranpatel11** [jayway/JsonPath#294]
isJson() is not returning true even if argument is valid json string.

```
private static boolean isJson(Object o) {
  if ((o == null) || (!(o instanceof String))) {
    return false;
  }
  String str = o.toString().trim();
  if (str.length() <= 1) {
    return false;
  }
  char c0 = str.charAt(0);
  char c1 = str.charAt(str.length() - 1);
  if (((c0 == '[') && (c1 == ']')) || ((c0 == '{') && (c1 == '}'))) {
    try {
      Configuration.defaultConfiguration().jsonProvider().parse(str);
      return false;   /// THIS SHOULD RETURN TRUE
    } catch (Exception e) {
      return false;
    }
  }
  return false;
}
```

**Comment by jlolling** [comment]
The parameter type Object often leads to problems. I suggest in this case also to accept JsonNode as input type.

**Comment by jochenberger** [comment]
@kiranpatel11 You're probably right. Can you provide a specific use case or a test?

**Comment by jochenberger** [comment]
Never mind

**Comment by kallestenflo** [comment]
Fixed by #330
